### PR TITLE
docs(headless commerce react samples): use react package

### PR DESCRIPTION
**Pull request by y-lakhdar**
### Refactoring Imports:
* Changed imports from `@coveo/headless/ssr-commerce` to `@coveo/headless-react/ssr-commerce` 

### State Management Improvements:
* Simplified state management by removing direct state handling and using custom hooks (`useBreadcrumbManager`, `useCart`, `useContext`, `useFacetGenerator`, `useInstantProducts`) for managing state in components.

### Component Simplification:
* Removed unnecessary props and state management logic from components since the controller is now fetched using react hooks.

### Page Component Enhancements:
* Better state management using `HydratedStateProvider` and `StaticStateProvider` for handling server-side rendering and client-side hydration. 

https://coveord.atlassian.net/browse/KIT-3699

### CODE REVIEW

File: packages/samples/headless-ssr-commerce/app/_components/recommendation-list.tsx [hunk 1]
```diff
@@ -1,29 +1,16 @@
...
+import {useRouter} from 'next/navigation';
+import {FunctionComponent} from 'react';
+import {usePopularBoughtRecs} from '../_lib/commerce-engine';
 
-export const Recommendations: FunctionComponent<RecommendationsProps> = ({
-  staticState,
-  controller,
-}) => {
-  const [state, setState] = useState(staticState);
+export const Recommendations: FunctionComponent = () => {
```
Comment (alexprudhomme) [inline packages/samples/headless-ssr-commerce/app/_components/recommendation-list.tsx:8]: Do you mind changing this to keep the same style across all of our react components.

```suggestion
export default function Recommendations() {

```

File: packages/samples/headless-ssr-commerce/app/layout.tsx [hunk 2]
```diff
@@ -2,8 +2,7 @@ import Link from 'next/link';
 
 export const metadata = {
   title: 'Headless SSR examples',
-  description:
-    'Examples of using framework agnostic @coveo/headless/ssr-commerce',
+  description: 'Examples of using framework @coveo/headless-react/ssr-commerce',
 };
 
 export default function RootLayout({children}: {children: React.ReactNode}) {
```
Comment (alexprudhomme) [inline packages/samples/headless-ssr-commerce/app/layout.tsx:8]: Can you also remove the link to the deleted recommendation page here. Thanks !

File: packages/samples/headless-ssr-commerce/app/_components/pages/listing-page.tsx [hunk 3]
```diff
@@ -1,29 +1,23 @@
...
 
 export default function ListingPage({
   staticState,
   navigatorContext,
-}: {
-  staticState: ListingStaticState;
-  navigatorContext: NavigatorContext;
-}) {
+  children,
+}: PropsWithChildren<ListingPageProps>) {
```
Comment (alexprudhomme) [inline packages/samples/headless-ssr-commerce/app/_components/pages/listing-page.tsx:20]: We could rename those components (search-page.tsx and listing-page.tsx) to something more like SearchProvider and ListingProvider. What do you think ?

File: packages/samples/headless-ssr-commerce/app/_components/pages/listing-page.tsx [hunk 4]
```diff
@@ -40,78 +34,28 @@ export default function ListingPage({
...
+      >
+        <>{children}</>
+      </listingEngineDefinition.HydratedStateProvider>
+    );
+  } else {
+    return (
+      <listingEngineDefinition.StaticStateProvider
+        controllers={staticState.controllers}
+      >
+        {/* // TODO: FIXME:  Type 'React.ReactNode' is not assignable to type 'import(".../node_modules/@types/react/index").ReactNode'.
```
Comment (fbeaudoincoveo) [inline packages/samples/headless-ssr-commerce/app/_components/pages/listing-page.tsx:55]: Add KIT-3701 in TODO comment

File: packages/samples/headless-ssr-commerce/app/_components/summary.tsx [hunk 5]
```diff
@@ -38,11 +20,12 @@ export default function Summary(props: ISummaryProps) {
       return null;
     }
 
-    return (
-      <span>
-        for <b>{state.query}</b>
-      </span>
-    );
+    // TODO: add query to summary state
```
Comment (fbeaudoincoveo) [inline packages/samples/headless-ssr-commerce/app/_components/summary.tsx:23]: Did you create a Jira for this? We should mention it in the TODO comment

File: packages/samples/headless-ssr-commerce/app/layout.tsx [hunk 6]
```diff
@@ -2,8 +2,7 @@ import Link from 'next/link';
 
 export const metadata = {
   title: 'Headless SSR examples',
-  description:
-    'Examples of using framework agnostic @coveo/headless/ssr-commerce',
+  description: 'Examples of using framework @coveo/headless-react/ssr-commerce',
```
Comment (fbeaudoincoveo) [inline packages/samples/headless-ssr-commerce/app/layout.tsx:5]:
```suggestion
  description: 'Examples of using @coveo/headless-react/ssr-commerce',
```

File: packages/samples/headless-ssr-commerce/package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@coveo/headless-ssr-commerce-samples",
-  "description": "Examples of framework agnostic @coveo/headless/ssr-commerce utils using Next.js app router",
+  "description": "Examples of framework @coveo/headless-react/ssr-commerce utils using Next.js app router",
```
Comment (fbeaudoincoveo) [inline packages/samples/headless-ssr-commerce/package.json:3]:
```suggestion
  "description": "Examples of @coveo/headless-react/ssr-commerce utils using Next.js app router",
```

**Review by fbeaudoincoveo — APPROVED** [review]
Nice!!

**Review by louis-bompart — APPROVED** [review]

**Review by alexprudhomme — APPROVED** [review]

**Pull request merged**